giftcertificate/dao/src/main/java/com/epam/esm/dao/impl/GiftCertificateJDBCTemplate.java

```diff
@@ -0,0 +1,227 @@
+package com.epam.esm.dao.impl;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import com.epam.esm.dao.GiftCertificateDAO;
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.entity.GiftCertificateMapper;
+import com.epam.esm.entity.Tag;
+import com.epam.esm.entity.TagMapper;
+import com.epam.esm.exception.DuplicateEntryDAOException;
+import com.epam.esm.exception.IdNotExistDAOException;
+import com.epam.esm.exception.RequestParamDAOException;
+import com.epam.esm.exception.UpdateDAOException;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.BatchPreparedStatementSetter;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+/**
+ * GiftSertificateJDBCTemplate - class for work with GiftCertificate
+ */
+@Repository
+public class GiftCertificateJDBCTemplate implements GiftCertificateDAO {
+
+    private static final Logger logger = Logger.getLogger(GiftCertificateJDBCTemplate.class);
+    private static final String SELECT_ALL_CERTIFICATES = "SELECT * FROM gift_db.gift_certificates order by id;";
+    private static final String SELECT_ALL_CERTIFICATES_WITH_SORT = "SELECT * FROM gift_db.gift_certificates order by %s %s;";
+    private static final String SELECT_CERTIFICATE_ID = "SELECT * FROM gift_db.gift_certificates where gift_certificates.id=?;";
+    private static final String INSERT_CERTIFICATE = "INSERT INTO gift_db.gift_certificates (name, description, price, duration, create_date, last_update_date) VALUES (?, ?, ?, ?, ?, ?);";
+    private static final String SELECT_TAGS_BY_CERTIFICATE_ID = "SELECT id, name FROM gift_db.gift_certificates_has_tags\n" +
+            "join gift_db.tags\n" +
+            "on gift_db.gift_certificates_has_tags.tags_id = gift_db.tags.id\n" +
+            "where gift_db.gift_certificates_has_tags.gift_certificates_id=?;";
+    private static final String CREATE_CERTIFICATE_HAS_TAG = "INSERT INTO `gift_db`.`gift_certificates_has_tags` (`gift_certificates_id`, `tags_id`) VALUES (?, ?);\n";
+    private static final String UPDATE_GIFT_CERTIFICATE = "UPDATE `gift_db`.`gift_certificates` SET `name` = ?, `description` = ?, `price` = ?, `duration` = ?, `last_update_date` = ? WHERE (`id` = ?);\n";
```

Comment:
Another good approach is not to use ?, but used named param for ex  SET name= :name, description = :desc.  Please try such approach.